### PR TITLE
Fixed prop type validation errors in info addon

### PR DESCRIPTION
**Pull request by Li0liQ**
Issue: https://github.com/storybooks/storybook/issues/1305

## What I did

Updated PropVal.js to pass its props to child elements in case of arrays and objects.

## How to test

Render a story using `addWithInfo` with a component that receives an array or an object as a prop. Open console to verify you no longer receive errors due to properties validation for PropVal component.

### CODE REVIEW

**Comment by Li0liQ** [comment]
Making eslint happy makes codebeat unhappy.
Should we say that `previewArray` and `previewObject` are functional react components, add prop types  validation for them and start passing all the arguments as a single `props` argument?

**Review by ndelangen — COMMENTED** [review]
@shilman @UsulPro Can you verify this PR is still valid considering #1501 ?

**Comment by UsulPro** [comment]
I'd prefer to completely change our approach for source generation.
We discuss possible solution in #1602

**Comment by danielduan** [comment]
This seems like a good bug fix that we should be merging into master? Doesn't seem like our Info refactor is gonna happen for 3.3.

**Review by danielduan — APPROVED** [review]

File: addons/info/src/components/PropVal.js [hunk 1]
```diff
@@ -51,13 +52,19 @@ function previewArray(val, maxPropArrayLength) {
   return <span style={valueStyles.array}>[{createFragment(items)}]</span>;
 }
 
-function previewObject(val, maxPropObjectKeys) {
+previewArray.propTypes = {
```
Comment (Hypnosphi) [inline addons/info/src/components/PropVal.js]: `previewArray` is never rendered as a component, it's only called as a function. So looks like those propTypes aren't really checked at any point. You should either remove them, or capitalize the name (`PreviewArray`) and replace `previewArray(props)` below with `<PreviewArray {...props} />`.

If you decide to do the latter, please note that `val` should rather be `PropTypes.array`, and don't disable all of the eslint rules — only the particular ones (`// eslint-disable react/forbid-prop-types`)

**Review by Hypnosphi — CHANGES_REQUESTED** [review]

**Comment by danielduan** [comment]
Gonna close this PR due to inactivity. Feel free to ping us if you'd like to keep working on it.

**Pull request closed without merging**